pkg/virt-operator/kubevirt_test.go

```diff
@@ -287,12 +287,23 @@ func (k *KubeVirtTestData) BeforeTest() {
 		return true, nil, nil
 	})
 	k.virtFakeClient.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
-		if action.GetVerb() == "list" && action.GetResource().Resource == apiinstancetype.ClusterPluralResourceName {
-			return true, &v1beta1.VirtualMachineClusterInstancetypeList{}, nil
-		}
-		if action.GetVerb() == "list" && action.GetResource().Resource == apiinstancetype.ClusterPluralPreferenceResourceName {
-			return true, &v1beta1.VirtualMachineClusterPreferenceList{}, nil
+		switch action.GetVerb() {
+		case "list":
+			switch action.GetResource().Resource {
+			case apiinstancetype.ClusterPluralResourceName:
+				return true, &v1beta1.VirtualMachineClusterInstancetypeList{}, nil
+			case apiinstancetype.ClusterPluralPreferenceResourceName:
+				return true, &v1beta1.VirtualMachineClusterPreferenceList{}, nil
+			}
+		case "create":
+			switch action.GetResource().Resource {
+			case apiinstancetype.ClusterPluralResourceName:
+				return true, &v1beta1.VirtualMachineClusterInstancetype{}, nil
+			case apiinstancetype.ClusterPluralPreferenceResourceName:
+				return true, &v1beta1.VirtualMachineClusterPreference{}, nil
+			}
```

Comment:
Why do we have this?

Reply:
The operator is now creating these resources by default. Alternatively we can disable deployment within these tests but as the default is to deploy I assumed we just wanted to cover these client calls instead?

Reply:
Right we want to test it, but why do we need to override default behavior of the fake client?

Reply:
Apologies got pulled away to work on something else, yeah I see what you mean and I can't recall why we ended up doing this with the original changes. I've added [67c1ee7](https://github.com/kubevirt/kubevirt/pull/12753/commits/67c1ee75ea0f5367e1460b2ae42e6c91d1905c19) that reworks this for these tests, can you take a look?

Reply:
@lyarwood Thanks, I will need more time this week but should get to it in 2-3 days

Reply:
[03febdaf1fa1192861be1975f9a14f597d2fdb31](https://github.com/kubevirt/kubevirt/pull/12753/commits/03febdaf1fa1192861be1975f9a14f597d2fdb31) now drops this entirely from the controller unit tests as the reconciler unit tests and functional tests already cover everything we need IMHO. Also dropped an unused test function in an addition commit.